src/event_emitter/mod.rs

```diff
@@ -0,0 +1,148 @@
+// Copyright 2020 Damir Jelić
+// Copyright 2020 The Matrix.org Foundation C.I.C.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::sync::Arc;
+
+use crate::events::{
+    call::{
+        answer::AnswerEvent, candidates::CandidatesEvent, hangup::HangupEvent, invite::InviteEvent,
+    },
+    direct::DirectEvent,
+    dummy::DummyEvent,
+    forwarded_room_key::ForwardedRoomKeyEvent,
+    fully_read::FullyReadEvent,
+    ignored_user_list::IgnoredUserListEvent,
+    key::verification::{
+        accept::AcceptEvent, cancel::CancelEvent, key::KeyEvent, mac::MacEvent,
+        request::RequestEvent, start::StartEvent,
+    },
+    presence::PresenceEvent,
+    push_rules::PushRulesEvent,
+    receipt::ReceiptEvent,
+    room::{
+        aliases::AliasesEvent,
+        avatar::AvatarEvent,
+        canonical_alias::CanonicalAliasEvent,
+        create::CreateEvent,
+        encrypted::EncryptedEvent,
+        encryption::EncryptionEvent,
+        guest_access::GuestAccessEvent,
+        history_visibility::HistoryVisibilityEvent,
+        join_rules::JoinRulesEvent,
+        member::MemberEvent,
+        message::{feedback::FeedbackEvent, MessageEvent},
+        name::NameEvent,
+        pinned_events::PinnedEventsEvent,
+        power_levels::PowerLevelsEvent,
+        redaction::RedactionEvent,
+        server_acl::ServerAclEvent,
+        third_party_invite::ThirdPartyInviteEvent,
+        tombstone::TombstoneEvent,
+        topic::TopicEvent,
+    },
+    room_key::RoomKeyEvent,
+    room_key_request::RoomKeyRequestEvent,
+    sticker::StickerEvent,
+    tag::TagEvent,
+    typing::TypingEvent,
+    CustomEvent, CustomRoomEvent, CustomStateEvent,
+};
+use crate::models::Room;
+
+use tokio::sync::Mutex;
+///
```

Comment:
This needs better docs, also an example of some sort, the one from the login.rs example might suffice.